packages/flutter_test/lib/src/mock_event_channel.dart

```diff
@@ -0,0 +1,59 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/services.dart';
+
+/// A mock stream handler for an [EventChannel] that mimics the native
+/// StreamHandler API.
+abstract class MockStreamHandler {
```

Comment:
Possible API improvement over what you have already:

If you added a `MockStreamHandler.inline` factory constructor that returns an `InlineMockStreamHandler`, users wouldn't have to know about the separate `InlineMockStreamHandler` class, which would improve discoverability.

If you did that, you could even make `InlineMockStreamHandler` private to this library. It would make autocomplete for the `TestDefaultBinaryMessenger` better in my opinion, as `InlineMockStreamHandler` wouldn't come up as a potential option when a user tries to type `MockStreamHandler.inline`.